app/models/api_token.rb

```diff
@@ -7,4 +7,20 @@ class APIToken < ApplicationRecord
   validates :lead_provider, presence: { message: "Lead provider must be specified" }
   validates :token, presence: { message: "Hashed token must be specified" }
   validates :token, uniqueness: { message: "Hashed token must be unique" }
+
+  class << self
+    def create_lead_provider_api_token!(lead_provider:, token: nil, description: nil)
+      description = "A lead provider token for #{lead_provider.name}" if description.nil?
+
+      create!(
+        lead_provider:,
+        token:,
+        description:
+      )
+    end
+
+    def find_lead_provider_api_token(token:)
```

Comment:
do we want to limit this by lead provider as well?

Reply:
I was in two minds about this, it felt a little pointless given we only support lead providers at the moment but maybe its worth it to avoid an inadvertent issue down the line if/when we add another foreign key 🤔 I'll add it in.